projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts

```diff
@@ -192,11 +197,22 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
             // handle scenario where last child row might not be in view
             // parent should scroll to child grid end
             const childContainer = this.grid.nativeElement.parentNode.parentNode;
-            const diff =
+            const diffBottom =
             childContainer.getBoundingClientRect().bottom - this.grid.rootGrid.nativeElement.getBoundingClientRect().bottom;
```

Comment:
I see that it takes into account only the grid boundaries, but will this scroll the actual child grid if the cell is partially visible in it?
Also this is not handled for `Home` button, both imo should either use default logic or override it this way.

Reply:
It is also used when goToLastBodyElement is called, which is executed when performShiftTabKey is called on summary row first cell. In that case the previous cell may not be visible (may be above the current visible viewport) so it may need to be scrolled up so that the target data cell is fully in view hence the additional handling.

Reply:
In this scenario though could't `navigateUp` be used instead?

Reply:
Not without modifying navigateUp to handle elements that are not row siblings (summary row is not in tbody but in footer) and since that function is mainly used for arrow key navigation (which should skip headers/footers and only check for sibling rows inside tbody) I'd rather not have summaries handling in it.

Reply:
Well it might not be the best place to handle it there because still would create some slight inconsistency when it will scroll but this is not big anyway.